### ISSUE TITLE
Upgrade to 3.2.1 and `/opt/conda/share/amrfinderplus/data/latestsh: 1: file: not found`

### ISSUE THREAD

**Issue by kristyhoran** [ncbi/amr#17]
Hi Arjun. We are using AMRfinderplus (brilliant by the way) in our public health genomics program at MDU in Australia @tseemann suggested I contact you. I am testing your recent upgrade to 3.2.1 and have run into a problem. I used the installation instructions for a conda install to build a singularity container and used `amrfinder -u` to ensure the database was updated. Inspection of the `latest` directory seems to show that everything is present. But when I try to run `amrfinder -n contigs.fa -o sample.out` I get the following 
```Running amrfinder -n contigs.fa -o sample.out
Software version: 3.2.1
Database version: 2019-10-30.1
AMRFinder translated nucleotide search with database /opt/conda/share/amrfinderplus/data/latestsh: 1: file: not found

*** ERROR ***
Cannot open file "/tmp/rgamiX.log"
``` 

Can you possibly point me in the right direction to fix this error? I greatly appreciate any help you can provide! Thanks in advance. 
Regards Kristy

**Comment by tseemann** [comment]
Is `amrfinder_update` the same as `amrfinder -u` ?

**Comment by vbrover** [comment]
bq. Is `amrfinder_update` the same as `amrfinder -u` ?
`amrfinder -u`  invokes `amrfinder_update`

While we are working on resolving this bug, you can proceed with using AMRFinder by adding the option `-d <directory with AMRFinder files>`.

**Comment by tseemann** [comment]
I think the `/opt/conda/share/amrfinderplus/data/latestsh: 1: file: not found` is really two lines:
```
/opt/conda/share/amrfinderplus/data/latest
sh: 1: file: not found```
```
And I suspect the `sh` error is from an `exec()` call in your C++ code.
Which I note you do not check any return codes for.

**Comment by evolarjun** [comment]
Hi Kristy,

Sorry for taking so long to address your issue. 

First off we should be checking return codes for exec() calls, but the error message is "file: not found".  As a short-cut we're relying on the free shell error message because exec() uses a shell to run the command (though the formatting and lack of detail on the message leaves something to be desired).

I have limited experience with containers. It looks like AMRFinderPlus is unable to read the database directory symlink. As Slava pointed out you could try either putting the actual database directory as an option to amrfinder e.g.:`-d /opt/conda/share/amrfinderplus/data/2019-10-30.1` or setting the AMRFINDER_DB variable to that directory before you call AMRFinderPlus (e.g.: `export AMRFINDER_DB=/opt/conda/share/amrfinderplus/data/2019-10-30.1`) I'm not sure why it is having trouble reading the symlink, but I haven't played with singularity containers at all. 

It may also be an effect of the hacky way I set up the database directories for conda installation (it was meant to be temporary, but like many hacks it has endured longer than it should).  We have plans to fix it, but they have not come to fruition yet. 

 I've only played with Docker containers, so I'll try downloading Singularity and creating a container and see if I can reproduce your issue.

Please let me know if manually setting the AMRFinderPlus database directory fixes your problem.

**Comment by andersgs** [comment]
@kristyhoran, @tseemann, and @evolarjun, I think I have cracked this nut. 

`file not found` refers to the UNIX tool `file` and not some file. Many of the base images distributed for containers are stripped of lots of tools in an effort to make them slimmer. 

So, adding something like this to your recipe fixes the problem (at least it fixed it for me):

```
apt-get --yes update && apt-get --yes upgrade && apt-get install --yes file && apt-get clean
```

Such a silly named program!!! 

Hope that helps. 

Cheers. A.

**Comment by tseemann** [comment]
@andersgs nice `find`!   (pun intended)

**Comment by mbi2gs** [comment]
Installing `find` worked for me in a Docker container. Definitely a nice "find" @andersgs! 

(I appreciated the pun @tseemann)

**Comment by evolarjun** [comment]
We had forgotten that there was a dependency on the file command. Sorry.

The file dependency will be removed in the next release which, tests willing, will be released soon.

**Comment by evolarjun** [comment]
Not released as soon as we had hoped, but version 3.6.4 (https://github.com/ncbi/amr/releases/tag/amrfinder_v3.6.4) contains the fix.
